--- lxdm-0.4.1/src/greeter.c	Mon Oct  7 11:23:27 2013
+++ lxdm-0.4.1/src/greeter.c	Mon Oct  7 11:33:23 2013
@@ -90,6 +90,11 @@
     printf("shutdown\n");
 }
 
+static void do_exitdm(void)
+{
+    printf("exitdm\n");
+}
+
 static char *get_session_lang(void)
 {
 	GtkTreeModel* model;
@@ -654,6 +659,10 @@
     g_signal_connect(item, "activate", G_CALLBACK(do_shutdown), NULL);
     gtk_menu_shell_append(GTK_MENU_SHELL(exit_menu), item);
 
+    item = gtk_image_menu_item_new_with_mnemonic( _("_Exit Display Manager") );
+    g_signal_connect(item, "activate", G_CALLBACK(do_exitdm), NULL);
+    gtk_menu_shell_append(GTK_MENU_SHELL(exit_menu), item);
+
     gtk_widget_show_all(exit_menu);
     g_signal_connect(exit_btn, "clicked", G_CALLBACK(on_exit_clicked), NULL);
 }
