--- clutter-gtk-1.0.4/po/Makefile.in.in	Thu May 14 09:05:32 2020
+++ clutter-gtk-1.0.4/po/Makefile.in.in.new	Thu May 14 09:05:54 2020
@@ -8,8 +8,8 @@
 # Please note that the actual code of GNU gettext is covered by the GNU
 # General Public License and is *not* in the public domain.
 #
-# Origin: gettext-0.18
-GETTEXT_MACRO_VERSION = 0.18
+# Origin: gettext-0.20
+GETTEXT_MACRO_VERSION = 0.20
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
